include/aws/cryptosdk/session.h

```diff
@@ -56,6 +56,22 @@ struct aws_cryptosdk_session;
 
 enum aws_cryptosdk_mode { AWS_CRYPTOSDK_ENCRYPT = 0x9000, AWS_CRYPTOSDK_DECRYPT = 0x9001 };
 
+/**
+ * Creates a new encryption or decryption session.
+ *
+ * @return The new session, or NULL on failure (in which case, an AWS error code is set)
+ *
+ * @param allocator The allocator to use for the session object and any temporary
+ *                  data allocated for the session
+ * @param mode The mode (AWS_CRYPTOSDK_ENCRYPT or AWS_CRYPTOSDK_DECRYPT) to start
+ *             in. This can be changed later with @ref aws_cryptosdk_session_reset
+ * @param keyring The keyring which will encrypt or decrypt data keys for this session.
+ *                This function uses a default CMM to link the session and keyring.
```

Comment:
If we are making the `default_cmm` to be implemented under the hood/ a private implementation detail, I dont think we should mention that in the public APIs documentation here.  This goes back to my lack of complete understanding on what is the motivation for this change.

Reply:
We are not making the ``default_cmm`` a private implementation detail. We're taking nothing out of the public API. We are providing convenience functions so that users don't have to worry about the details of the default CMM if they don't want to.

We need to keep the CMM interface public, because we still want users to have the ability to create their own CMMs and plug whichever ones they want into sessions.

Furthermore, if users want to change the algorithm suite from the default one, they currently need to manually construct a default CMM to do that. Most users won't need this functionality, but we're not going to take it away.

(Note that we could possibly add functionality to set the algorithm ID in the session, but that would only be for a single encryption or decryption. Some users might still want the functionality to have a CMM persistent across many encryptions and decryptions to always use one of the alternative algorithm suites.)

The tl;dr here is that many users don't need to worry about default CMMs and we're giving them a simpler interface where they can just create a keyring and session and never think about CMMs. But power users still need access to the CMMs.

Reply:
Alright, thank-you for the explanation.